packages/flutter_tools/test/general.shard/generate_localizations_test.dart

```diff
@@ -1766,6 +1766,39 @@ import 'output-localization-file_en.dart' deferred as output-localization-file_e
           )),
         );
       });
+
+      testWithoutContext('should not fail attempting to generate a plural message with an incorrect message', () {
```

Comment:
According to our documentation - flutter.dev/go/i18n-user-guide, "Messages With Plurals" - the `other` variation is required. As @ened [pointed out](https://github.com/flutter/flutter/pull/90096/files#r708956126), `Intl.plural` may accept this and assume that `other` is an empty string, however that seems likely to mask an error.

It might be best to leave the plural test out for now, since this PR was really about fixing up `select`.

Also: a more ambitious update to select/plurals has been proposed: https://github.com/flutter/flutter/issues/86906